zenoh/src/net/routing/hat/client/mod.rs

```diff
@@ -275,10 +275,8 @@ impl HatBaseTrait for HatCode {
         _expr: &mut RoutingExpr,
     ) -> bool {
         src_face.id != out_face.id
-            && match (src_face.mcast_group.as_ref(), out_face.mcast_group.as_ref()) {
-                (Some(l), Some(r)) => l != r,
-                _ => true,
-            }
+            && out_face.mcast_group.is_none()
```

Comment:
why this change?

Reply:
It's to avoid sending data to a multicast group when in client mode.
This is also to align with zenoh-pico where joining a multicast group can only be done when in peer mode.